admin/jobs/river/biller_event_handlers.go

```diff
@@ -308,3 +309,58 @@ func (w *PaymentFailedGracePeriodCheckWorker) checkFailedInvoicesForOrg(ctx cont
 	}
 	return hasOverdue, nil
 }
+
+type PlanCacheUpdateArgs struct {
+	BillingCustomerID string
+}
+
+func (PlanCacheUpdateArgs) Kind() string { return "plan_cache_update" }
+
+type PlanCacheUpdateWorker struct {
+	river.WorkerDefaults[PlanCacheUpdateArgs]
+	admin *admin.Service
+}
+
+func (w *PlanCacheUpdateWorker) Work(ctx context.Context, job *river.Job[PlanCacheUpdateArgs]) error {
+	org, err := w.admin.DB.FindOrganizationForBillingCustomerID(ctx, job.Args.BillingCustomerID)
+	if err != nil {
+		if errors.Is(err, database.ErrNotFound) {
+			// org got deleted, ignore
+			return nil
+		}
+		return fmt.Errorf("failed to find organization of billing customer id %q: %w", job.Args.BillingCustomerID, err)
+	}
+
+	// something related to plan changed, just fetch the latest plan from the biller
+	sub, err := w.admin.Biller.GetActiveSubscription(ctx, org.BillingCustomerID)
+	if err != nil {
+		if errors.Is(err, billing.ErrNotFound) {
+			return nil
+		}
+		return fmt.Errorf("failed to get subscriptions for org %q: %w", org.Name, err)
+	}
+
+	_, err = w.admin.DB.UpdateOrganization(ctx, org.ID, &database.UpdateOrganizationOptions{
+		Name:                                org.Name,
+		DisplayName:                         org.DisplayName,
+		Description:                         org.Description,
+		LogoAssetID:                         org.LogoAssetID,
+		CustomDomain:                        org.CustomDomain,
+		QuotaProjects:                       org.QuotaProjects,
+		QuotaDeployments:                    org.QuotaDeployments,
+		QuotaSlotsTotal:                     org.QuotaSlotsTotal,
+		QuotaSlotsPerDeployment:             org.QuotaSlotsPerDeployment,
+		QuotaOutstandingInvites:             org.QuotaOutstandingInvites,
+		QuotaStorageLimitBytesPerDeployment: org.QuotaStorageLimitBytesPerDeployment,
+		BillingCustomerID:                   org.BillingCustomerID,
+		PaymentCustomerID:                   org.PaymentCustomerID,
+		BillingEmail:                        org.BillingEmail,
+		CreatedByUserID:                     org.CreatedByUserID,
+		CachedPlanDisplayName:               &sub.Plan.DisplayName,
```

Comment:
Is it guaranteed to have a display name? If it may be empty (e.g. in case we forgot to set one in Orb), it should perhaps fall back to the unique name or ID instead (so it doesn't look like no plan is assigned).

Reply:
Plan display name is not set in orb at all, its generated logically as per product requirement here -https://github.com/rilldata/rill/blob/main/admin/billing/orb.go#L636